### ISSUE TITLE
depwarns please

### ISSUE THREAD

**Issue by bjarthur** [JuliaMath/Polynomials.jl#246]
i really like the new API, whereby one creates a polynomial with `Polynomial` instead of `Poly` and then evaluates it as a functor instead of calling `polyeval`.  thanks for that!

however, upgrading is more difficult than it needs to be.  one has to read the documentation!  gheesh.  would be more discoverable were there better deprecation warnings.  currently, it just says `Poly` is deprecated and `polyeval` is not found.  how about saying "use `Polynomial` instead of `Poly`" and "use `Polynomial(...)(...)` instead of `polyeval`" ?

```
julia> using Polynomials.PolyCompat

help?> Poly
search: Poly poly polyval polyint polyfit polyder PolyCompat @polly evalpoly @evalpoly

  Poly{T}

  Type of polynomial to support legacy code. Use of this type is not encouraged.

  This type provides support for poly, polyval, polyder, and polyint to support older code. It
  should not be used for a new code base. Call using Polynomials.PolyCompat to enable this
  module.

help?> polyeval
search:

Couldn't find polyeval
Perhaps you meant polyval, padeval or polyder
  No documentation found.

  Binding polyeval does not exist.
```

**Comment by jverzani** [comment]
Yeah, this wasn't the most graceful. The basic idea was legacy code would need that extra `using Polynomials.PolyCompat` and new code would use the newer names for the types. The `polyval` (no `e`, as it was borrowed from MATLAB, presumably) doesn't even have a doc string. There was a very short deprecation phase where some warnings were issued, but I was in a rush to get 1.0 out the door and didn't have those around too long. If you had some specific wording changes in `Poly.jl` that would be most welcome.
